### PR TITLE
[Antithesis | Range Queries] Part 5: Random Shuffle Workflow

### PR DESCRIPTION
**Pull request by jeremyk-91**
## General
**Before this PR**: The random workflow only does reads, puts or deletes.

**After this PR**:

==COMMIT_MSG==
The random workflow now also does row-column-range-scans.
==COMMIT_MSG==

**Priority**: P2

**Concerns / possible downsides (what feedback would you like?)**: Not much.

**Is documentation needed?**: No

## Compatibility
Antithesis workload server.

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**: Not much

**What was existing testing like? What have you done to improve it?**: Added relevant tests

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**: N/A

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**: N/A

## Execution
Antithesis workload server.

## Scale
Antithesis workload server.

## Development Process
**Where should we start reviewing?**: It's small

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**: N/A

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/RandomWorkflows.java [hunk 1]
```diff
@@ -111,8 +117,37 @@ private List<DeleteTransactionAction> generateDeleteActions() {
                     .collect(Collectors.toList());
         }
 
+        private List<RowColumnRangeReadTransactionAction> generateRowColumnRangeReadActions() {
+            return IntStream.range(0, random.nextInt(workflowConfiguration.maxRowRangeScans() + 1))
```
Comment (mdaudali) [inline atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/RandomWorkflows.java:121]: no action: This is fun - two separate exclusive endpoints (next int goes to X+1 -> max is X, then range goes up to X - 1 leaving X values). Nothing required, was just fun to double check that it was correct.

**Review by mdaudali — APPROVED** [review]
lgtm, 1 q

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/RandomWorkflowConfiguration.java [hunk 2]
```diff
@@ -47,8 +47,18 @@ default Integer maxDeletes() {
     }
 
     @Value.Default
-    default Integer maxCells() {
-        return 10;
+    default Integer maxRowRangeScans() {
+        return 3;
```
[the remark this answers is not in the document]
Comment (jeremyk-91) [reply atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/RandomWorkflowConfiguration.java:51]: The general intuition the Antithesis consultants provided was generally to go small, as they provide sufficient randomness to explore the ranges of interactions with not-too-large-N. There's a tradeoff in that choosing a number that's too small would prevent issues only visible when that number of requests were made. One could argue `1` might be sufficient, but I wasn't too sure (e.g. things around local writes etc) so went with `3`.

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/RandomWorkflows.java (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/RandomWorkflows.java:121]: Yeah - in the original PR there was a bit of funky discussion around this as well :)

Comment (jeremyk-91) [reply atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/RandomWorkflows.java:121]: basically, annoyingly you need to pick from N+1 outcomes, and given the APIs this drops out

**Pull request merged**